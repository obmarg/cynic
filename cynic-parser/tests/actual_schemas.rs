macro_rules! schema_tests {
    ($name: ident, $path: literal) => {
        mod $name {
            use similar_asserts::assert_eq;

            #[allow(non_upper_case_globals)]
            const SCHEMA: &str = include_str!($path);

            #[test]
            fn snapshot() {
                let parsed = cynic_parser::parse_type_system_document(SCHEMA)
                    .map_err(|error| error.to_report(SCHEMA))
                    .unwrap();
                insta::assert_snapshot!(parsed.to_sdl_pretty());
            }

            #[test]
            fn double_roundtrip() {
                let parsed = cynic_parser::parse_type_system_document(SCHEMA)
                    .map_err(|error| error.to_report(SCHEMA))
                    .unwrap();
                let sdl = parsed.to_sdl_pretty();

                let reparsed = cynic_parser::parse_type_system_document(&sdl)
                    .map_err(|error| error.to_report(&sdl))
                    .unwrap();

                assert_eq!(sdl, reparsed.to_sdl_pretty());
            }
        }
    };
}

schema_tests!(github, "../../schemas/github.graphql");

schema_tests!(books, "../../schemas/books.graphql");

schema_tests!(graphql_jobs, "../../schemas/graphql.jobs.graphql");

schema_tests!(raindancer, "../../schemas/raindancer.graphql");

schema_tests!(simple, "../../schemas/simple.graphql");

schema_tests!(starwars, "../../schemas/starwars.schema.graphql");

schema_tests!(test_cases, "../../schemas/test_cases.graphql");
