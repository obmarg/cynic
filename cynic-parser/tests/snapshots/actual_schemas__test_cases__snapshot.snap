---
source: cynic-parser/tests/actual_schemas.rs
expression: parsed.to_sdl()
---
schema {
  query: Foo
}

type Foo {
  _: Boolean
  self: Boolean
  super: Boolean
  crate: Boolean
  async: Boolean
  bar(id: UUID!): Bar
  fieldWithKeywordArg(where: Int): [Int!]!
  recursiveInputField(
    recursive: SelfRecursiveInput
    recursive2: RecursiveInputParent
  ): Boolean
  fieldWithStringArg(input: String!): Int
}

type Bar {
  id: UUID!
  name: String
}

scalar UUID

enum States {
  OPEN
  CLOSED
  DELETED
}

input SelfRecursiveInput {
  recurse: SelfRecursiveInput
}

input RecursiveInputParent {
  recurse: RecursiveInputChild
}

input RecursiveInputChild {
  recurse: RecursiveInputParent
}

type FlattenableEnums {
  states: [States]
}

